tests/test_mkieee.F90

```diff
@@ -29,6 +29,11 @@ program test_mkieee
      print *, i, a(i), b(i)
      if (abs(a(i) - b(i)) .gt. .00001) stop 20
   end do
+
+  call mkieee(a, rieee, num)
+
+  if (atemp .lt. 0.0 .and. n .ne. -1) stop 30
```

Comment:
I don't even understand how this compiled, because atemp is not defined and we have implicit none. So how come we're not getting a variable not found error?

Reply:
What you need to do is extend array a by adding some negative numbers, and then make sure we get the correct result from mkieee, which means check the values returned in rieee.

Basically, mkieee takes array a, with length num, and then fills array rieee with the ieee versions of those numbers.